servicetalk-concurrent-jdkflow/src/main/java/io/servicetalk/concurrent/jdkflow/JdkFlowAdapters.java

```diff
@@ -0,0 +1,200 @@
+        private final PublisherSource<T> source;
+
+        PublisherToFlowPublisher(final Publisher<T> source) {
+            this.source = toSource(requireNonNull(source));
+        }
+
+        @Override
+        public void subscribe(final Subscriber<? super T> subscriber) {
+            source.subscribe(new FlowSubscriber<>(subscriber));
+        }
+    }
+
+    private static final class PublisherSourceToFlowPublisher<T> implements java.util.concurrent.Flow.Publisher<T> {
+        private final PublisherSource<T> source;
+
+        PublisherSourceToFlowPublisher(final PublisherSource<T> source) {
+            this.source = requireNonNull(source);
+        }
+
+        @Override
+        public void subscribe(final Subscriber<? super T> subscriber) {
+            source.subscribe(new FlowSubscriber<>(subscriber));
+        }
+    }
+
+    private static final class FlowSubscriber<T> implements PublisherSource.Subscriber<T> {
+        private final Subscriber<? super T> subscriber;
+
+        FlowSubscriber(final Subscriber<? super T> subscriber) {
+            this.subscriber = requireNonNull(subscriber);
+        }
+
+        @Override
+        public void onSubscribe(final PublisherSource.Subscription subscription) {
+            subscriber.onSubscribe(new Subscription() {
+                @Override
+                public void request(final long n) {
+                    subscription.request(n);
+                }
+
+                @Override
+                public void cancel() {
+                    subscription.cancel();
+                }
+            });
+        }
+
+        @Override
+        public void onNext(final T next) {
+            subscriber.onNext(next);
+        }
+
+        @Override
+        public void onError(final Throwable t) {
+            subscriber.onError(t);
+        }
+
+        @Override
+        public void onComplete() {
+            subscriber.onComplete();
+        }
+    }
+
+    private static final class FlowPublisherToPublisher<T> extends Publisher<T> {
+        private final java.util.concurrent.Flow.Publisher<T> source;
+
+        FlowPublisherToPublisher(final java.util.concurrent.Flow.Publisher<T> source) {
+            this.source = requireNonNull(source);
+        }
+
+        @Override
+        protected void handleSubscribe(final PublisherSource.Subscriber<? super T> subscriber) {
+            source.subscribe(new FlowToSTSubscriber<>(subscriber));
+        }
+    }
+
+    private static final class FlowToSTSubscription implements PublisherSource.Subscription {
+        private final Subscription s;
+
+        FlowToSTSubscription(final Subscription s) {
+            this.s = s;
+        }
+
+        @Override
+        public void request(final long n) {
+            s.request(n);
+        }
+
+        @Override
+        public void cancel() {
+            s.cancel();
+        }
+    }
+
+    private static final class FlowToSTSubscriber<T> implements Subscriber<T> {
```

Comment:
`StToFlowSubscriber` is a better name here as we are converting a ServiceTalk `Subscriber` to Flow `Subscriber`.

Reply:
Thats right. Fixing it.

Reply:
Since the constructor accepts ST's `PublisherSource.Subscriber` and implements Flow's `Subscriber` I think it is more natural to think about it as converting an ST subscriber to Flow subscriber, hence the name suggestion.

Reply:
ok